test/end2end_test.go

```diff
@@ -6880,6 +6880,170 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	conn := &connWrapperWithCloseCh{Conn: rawConn, CloseCh: testutils.NewChannel()}
+	c.connections = append(c.connections, *conn)
+	authInfo := &authInfoWithConn{CommonAuthInfo: credentials.CommonAuthInfo{SecurityLevel: credentials.NoSecurity}, connection: conn}
+	return conn, authInfo, nil
+}
+func (c *transportRestartCheckCreds) Info() credentials.ProtocolInfo {
+	return credentials.ProtocolInfo{}
+}
+func (c *transportRestartCheckCreds) Clone() credentials.TransportCredentials {
+	return &transportRestartCheckCreds{}
+}
+func (c *transportRestartCheckCreds) OverrideServerName(s string) error {
+	return nil
+}
+
+type streamWithPeer struct {
+	stream testpb.TestService_FullDuplexCallClient
+	peer   *peer.Peer
+	id     string
+	conn   net.Conn
+}
+
+func (s *streamWithPeer) sendAndReceive() error {
+	if err := s.stream.Send(&testpb.StreamingOutputCallRequest{}); err != nil {
+		return fmt.Errorf("sending on %s: %v", s.id, err)
+	}
+	if _, err := s.stream.Recv(); err != nil {
+		return fmt.Errorf("receiving on %s: %v", s.id, err)
+	}
+	return nil
+}
+
+// Tests that the client transport drains and restarts when next stream ID exceeds
+// MaxStreamID. This test also verifies that subsequent RPCs use a new client
+// transport and the old transport is closed.
+func (s) TestClientTransportRestartsAfterStreamIDExhausted(t *testing.T) {
+	// Set the transport's MaxStreamID to 5 to cause connection to drain after 2 RPCs.
+	originalMaxStreamID := transport.MaxStreamID
+	transport.MaxStreamID = 5
+	defer func() {
+		transport.MaxStreamID = originalMaxStreamID
+	}()
+
+	ss := &stubserver.StubServer{
+		FullDuplexCallF: func(stream testpb.TestService_FullDuplexCallServer) error {
+			for {
+				if _, err := stream.Recv(); err != nil {
+					return err
+				}
+				if err := stream.Send(&testpb.StreamingOutputCallResponse{}); err != nil {
+					return err
+				}
+			}
+		},
+	}
+
+	creds := &transportRestartCheckCreds{}
+	if err := ss.Start(nil, grpc.WithTransportCredentials(creds)); err != nil {
+		t.Fatalf("starting stubServer: %v", err)
+	}
+	defer ss.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	var streams []streamWithPeer
+
+	// Setting up 3 streams and calling Send() and Recv() once on each.
+	for i := 0; i < 3; i++ {
+		var p peer.Peer
+		s, err := ss.Client.FullDuplexCall(ctx, grpc.Peer(&p))
+		if err != nil {
+			t.Fatalf("creating FullDuplex stream: %v", err)
+		}
+		streamID := fmt.Sprintf("stream %v", i)
+		sp := streamWithPeer{stream: s, peer: &p, id: streamID}
+		streams = append(streams, sp)
+
+		if err := sp.sendAndReceive(); err != nil {
+			t.Fatal(err)
+		}
+	}
+	// Verifying only 2 connections are created so far.
+	if len(creds.connections) != 2 {
+		t.Fatalf("number of connections created: %v, want: 2", len(creds.connections))
```

Comment:
The strings passed to `t.Error/Fatal` are not considered error strings and hence the capitalization recommendation does not apply to them. They should be capitalized where appropriate. 

Applies to here and everywhere else in this PR.

Reply:
noted. updating this